lib/constants/ranks.ts

```diff
@@ -31,16 +31,29 @@ export enum Ranks {
   Feast = 'FEAST',
 
   /**
-   * Memorials are either obligatory or optional; their observance is integrated into
-   * the celebration of the occurring weekday in accordance with the norms set forth in the
-   * General Instruction of the Roman Missal and of the Liturgy of the Hours.
-   * Obligatory Memorials which fall on weekdays of Lent may only be celebrated as
-   * Optional Memorials.
-   * If several Optional Memorials are inscribed in the Calendar on the same day, only
-   * one may be celebrated, the others being omitted. (UNLY #14)
+   * Obligatory Memorials are liturgical commemorations of saints, events, or aspects of the faith.
+   * Their observance is mandatory and integrated into the celebration of the occurring weekday,
+   * following the liturgical norms outlined in the General Instruction of the Roman Missal and the
+   * Liturgy of the Hours.
+   * When an Obligatory Memorial falls on a weekday during the liturgical season of Lent or a
+   * privileged weekday of Advent, it must only be celebrated as an Optional Memorial, as Lent and
```

Comment:
Would you think that in these cases (during Lent and from 17 Dec), we should define a new `rank` type as `COMMEMORATION`? At first glance, I think it's a good idea  👍  Especially since the precedence property also provides useful information about the celebration.

Reply:
Well, defining a new `rank` for commemorations is a good idea, however, I am not sure if that is that necessary (there is no such type in the GIRM). IMHO adding a `LiturgicalDay` (celebration) property (e.g. `isCommemoration` of a `boolean` type) could suffice.

Reply:
I'm not very enthusiastic about adding an `isCommemoration` property. 
Especially since memories, feasts, and other ranks are also commemorations, having an `isCommemoration: false` property for these celebrations seems counterintuitive to me. 

The term "commemoration" (for a rank) remains quite generic, so in our case here, it seemed appropriate (i.e. we specify that this element remains a commemoration, even though it no longer holds the official rank of a memory).

Let's save this discussion for later and for a dedicated topic because I believe we need to reexamine the missal instructions on this issue and let this idea mature a bit.

Reply:
> I'm not very enthusiastic about adding an `isCommemoration` property.
> Especially since memories, feasts, and other ranks are also commemorations, having an `isCommemoration: false` property for these celebrations seems counterintuitive to me.

You might be right.

I was referring to this section of the GIRM:

> 

On the weekdays of Advent from December 17 to December 24, on days within the Octave of Christmas, and on the weekdays of Lent, except Ash Wednesday and during Holy Week, the Mass for the current liturgical day is to be used; but the collect may be taken from a memorial which happens to be listed in the General Calendar for that day, except on Ash Wednesday and during Holy Week. On weekdays of the Easter Season, memorials of Saints may rightly be celebrated fully.

Whenever a proper celebration is prohibited by this and only the _collect may be take from a memorial_, I considered this a _commemoration_. I have no idea where I found this term though.

> Let's save this discussion for later and for a dedicated topic because I believe we need to reexamine the missal instructions on this issue and let this idea mature a bit.

Agreed. :+1: